client/components/AddSponsor.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import fetch from 'isomorphic-fetch';
+
+import {
+  Form,
+  Input,
+  ResponseDiv,
+  SubmitBtn,
+} from 'client/styles/components/AddSponsor';
+import {
+  IAddSponsorProps,
+  ISponsorData,
+} from 'client/interfaces/components/AddSponsor';
+
+const AddSponsor: React.FC<IAddSponsorProps> = ({ eventId, chapterId }) => {
+  const [isSubmitting, setSubmitting] = React.useState(false);
+  // we specify type as food as state is only updated on field change. Since field
+  // defaults to food and the user doesn't change it, state would be empty
+  const [values, setValues] = React.useState<ISponsorData>({
```

Comment:
Have we discussed using `useReducer` instead of `useState` with an object for this? For this specific form it's OK, but if we had more fields maybe `useReducer` hook would make it more readable, since we're just implementing our own reducer later in the `handleChange`?

Reply:
I don't think this has been discussed as a pattern we want to use - feel free to create a new issue for it!